core/src/main/java/io/grpc/internal/InternalSubchannel.java

```diff
@@ -313,20 +270,21 @@ public void run() {
         "TRANSIENT_FAILURE ({0}). Will reconnect after {1} ns",
         printShortStatus(status), delayNanos);
     Preconditions.checkState(reconnectTask == null, "previous reconnectTask is not done");
-    reconnectCanceled = false;
-    reconnectTask = scheduledExecutor.schedule(
+    reconnectTask = syncContext.schedule(
         new LogExceptionRunnable(new EndOfCurrentBackoff()),
         delayNanos,
-        TimeUnit.NANOSECONDS);
+        TimeUnit.NANOSECONDS,
```

Comment:
We probably don't need to use `LogExceptionRunnable` anymore. It was necessary because the runnable was scheduled on `ScheduledExecutorService` which doesn't handle exceptions as we would like to (6a04022d7aceb73d9cf8096b107930eac0f6c477)

Reply:
right. syncContext is already handling exception.